### PR TITLE
Fix regex match -> exec bug in amp-img

### PR DESCRIPTION
**Pull request by cathyxz**
Fixes https://github.com/ampproject/amphtml/issues/15769

### CODE REVIEW

File: builtins/amp-img.js [hunk 1]
```diff
@@ -208,13 +208,17 @@ export class AmpImg extends BaseElement {
...
    */
   guaranteeSrcForSrcsetUnsupportedBrowsers_() {
-    if (!this.img_.hasAttribute('src') && 'srcset' in this.img_ == false) {
-      const srcset = this.element.getAttribute('srcset');
-
-      const srcseturl = /\S+/.match(srcset);
-      if (srcseturl) {
-        this.img_.setAttribute('src', srcseturl[0]);
-      }
+    if (this.img_.hasAttribute('src') || 'srcset' in this.img_ == false) {
```
Comment (nainar) [inline builtins/amp-img.js]: This isn't the deMorgan's equivalent of the deleted statement.

File: builtins/amp-img.js [hunk 2]
```diff
@@ -208,10 +208,14 @@ export class AmpImg extends BaseElement {
...
       const srcset = this.element.getAttribute('srcset');
-
-      const srcseturl = /\S+/.match(srcset);
+      const matches = /\S+/.exec(srcset);
+      if (matches == null) {
+        return;
+      }
+      const srcseturl = matches[0];
       if (srcseturl) {
         this.img_.setAttribute('src', srcseturl[0]);
```
Comment (jridgewell) [inline builtins/amp-img.js]: This sets src to a char.

Comment (cathyxz) [reply builtins/amp-img.js]: 0.0 oh god. Good catch! Fixed and testing that now.

Comment (cathyxz) [reply builtins/amp-img.js]: Tested this on IE. It works.

**Comment by cathyxz** [comment]
@jridgewell PTAL? ^.^

File: builtins/amp-img.js [hunk 3]
```diff
@@ -208,12 +208,16 @@ export class AmpImg extends BaseElement {
...
     if (!this.img_.hasAttribute('src') && 'srcset' in this.img_ == false) {
       const srcset = this.element.getAttribute('srcset');
-
-      const srcseturl = /\S+/.match(srcset);
+      const matches = /\S+/.exec(srcset);
+      if (matches == null) {
+        return;
+      }
+      const srcseturl = matches[0];
       if (srcseturl) {
```
Comment (jridgewell) [inline builtins/amp-img.js]: Nit: this check can never be false.

File: builtins/amp-img.js [hunk 4]
```diff
@@ -208,12 +208,16 @@ export class AmpImg extends BaseElement {
    * @private
    */
   guaranteeSrcForSrcsetUnsupportedBrowsers_() {
+    // The <img> tag does not have a src and does not support srcset
     if (!this.img_.hasAttribute('src') && 'srcset' in this.img_ == false) {
       const srcset = this.element.getAttribute('srcset');
-
-      const srcseturl = /\S+/.match(srcset);
+      const matches = /\S+/.exec(srcset);
```
Comment (jridgewell) [inline builtins/amp-img.js]: I'm not seeing any tests for this.

**Comment by cathyxz** [comment]
Not yet sure if the adding IE to `saucelabs_lite` thing will work because IE does not seem to support chai-as-promised (at least not in my local runs). If this doesn't work, we may have to write this as an integration test instead.

**Comment by cathyxz** [comment]
Enabling IE on unit tests breaks too many unit tests to be feasible, test rewritten to be an integration test instead.

**Pull request merged**